components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManagerImpl.java

```diff
@@ -531,6 +531,39 @@ private void validateResourceCreateRequest(String resourceTypeName, ResourceAdd
         }
     }
 
+    private void validateResourceCreateRequest(String resourceTypeName, Resource resource)
+            throws ConfigurationManagementException {
+
+        if (StringUtils.isEmpty(resourceTypeName) || StringUtils.isEmpty(resource.getResourceName())) {
+            throw handleClientException(ERROR_CODE_RESOURCE_ADD_REQUEST_INVALID, null);
+        }
+        if (isResourceExists(resourceTypeName, resource.getResourceName())) {
+            throw handleClientException(ERROR_CODE_RESOURCE_ALREADY_EXISTS, resource.getResourceName());
+        }
+        if (CollectionUtils.isNotEmpty(resource.getFiles())) {
+            List<ResourceFile> files = resource.getFiles();
+            for (ResourceFile file : files) {
+                if (StringUtils.isEmpty(file.getId())) {
+                    file.setId(generateUniqueID());
+                }
+                if (StringUtils.isEmpty(file.getName())) {
+                    String fileIdentifiers = String.format("Resource type: %s, resourceName: %s", resourceTypeName,
+                            resource.getResourceName());
+                    throw handleClientException(ERROR_CODE_FILE_IDENTIFIERS_REQUIRED, fileIdentifiers);
+                }
+                if (file.getInputStream() == null) {
+                    throw handleClientException(ERROR_CODE_FILE_IDENTIFIERS_REQUIRED, "fileStream is invalid.");
```

Comment:
"fileStream is invalid." -> "Filestream is Emprty."

Reply:
Fixed in d8a058a